src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,186 @@
+import java.time.LocalDate;
+
+public class Duke {
+
+    // Attributes of a Duke object
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+    private final static String FILEDESTINATION = "./src/main/duke.txt";
+
+    /**
+     * The constructor for the Duke class
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage(FILEDESTINATION);
+        taskList = new TaskList();
+    }
+
+    /**
+     * The method to run the Duke bot
+     * @throws DukeException exception to be thrown regarding DukeException
+     * @throws IOException exception to be thrown regarding IOException
+     */
+    public void run() throws DukeException, IOException {
+        storage.loadTasks();
+        ui.greet();
+        ui.showTaskList();
+
+        // Boolean value to check if user wants to leave
+        boolean isQuit = false;
+
+        while (!isQuit) {
+            try {
+                String userCommand = ui.getCommand();
+                if (userCommand.equals("bye")) {
+                    isQuit = true;
+                    ui.bye();
+                } else if (userCommand.equals("list")) {
+                    ui.showTaskList();
+                } else {
+                    // Get all the words the user has typed
+                    String[] words = userCommand.split(" ");
+                    // Check if user wants to mark task
+                    if (Parser.isMarkTask(words)) {
+                        int taskNumber = Integer.parseInt(words[1]);
+                        // Check if user enters a number out of range
+                        if (taskNumber < 0 || taskNumber > TaskList.taskArrayList.size()) {
+                            throw new DukeException("Number out of range!");
+                        } else {
+                            ui.markTaskDoneAndPrintOutput(taskNumber);
+                        }
+                        // Checks if user wants to unmark task
+                    } else if (Parser.isUnmarkTask(words)) {
+                        int taskNumber = Integer.parseInt(words[1]);
+                        // Check if user enters a number out of range
+                        if (taskNumber < 0 || taskNumber > TaskList.taskArrayList.size()) {
+                            throw new DukeException("Number out of range!");
+                        } else {
+                            ui.markTaskNotDoneAndPrintOutput(taskNumber);
+                        }
+                        // Check if user wants to delete a task
+                    } else if (Parser.isDeleteTask(words)) {
+                        int taskNumber = Integer.parseInt(words[1]);
+                        if (taskNumber < 0 || taskNumber >TaskList.taskArrayList.size()) {
+                            throw new DukeException("Number out of range!");
+                        } else {
+                            ui.markTaskDeletedAndPrintOutput(taskNumber);
+                        }
```

Comment:
Consider separating the logical units that you have indicated well with the comments, with a blank line for better readability